### PR TITLE
[refactor]remove seek block in segmentIterator

### PR DESCRIPTION
**Pull request by wangbo**
# Proposed changes
remove ```std::unique_ptr<RowBlockV2> _seek_block``` from in ```SegmentIterator``` , make prepare for subsequent deletion of non vectorized code.
Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.74 seconds
 load time: 637 seconds
 storage size: 17123651617 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221227165226_clickbench_pr_69635.html

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -571,11 +569,13 @@ Status SegmentIterator::_seek_and_peek(rowid_t rowid) {
         RETURN_IF_ERROR(_seek_columns(_seek_schema->column_ids(), rowid));
     }
     size_t num_rows = 1;
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:571]: num_rows is useless

File: be/src/olap/rowset/segment_v2/segment_iterator.h [hunk 2]
```diff
@@ -227,7 +281,10 @@ class SegmentIterator : public RowwiseIterator {
     std::unique_ptr<Schema> _seek_schema;
     // used to binary search the rowid for a given key
     // only used in `_get_row_ranges_by_keys`
-    std::unique_ptr<RowBlockV2> _seek_block;
+    vectorized::MutableColumns _seek_block;
+
+    //todo(wb) remove this field after Rowcursor is removed
+    vectorized::MutableColumns _short_key;
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/segment_iterator.h:287]: _short_key -> _seek_key seems better？it's not **short** actually.

File: be/src/olap/rowset/segment_v2/segment_iterator.h [hunk 3]
```diff
@@ -166,6 +167,59 @@ class SegmentIterator : public RowwiseIterator {
 
     void _update_max_row(const vectorized::Block* block);
 
+    // todo(wb) remove this method after RowCursor is removed
+    void _convert_rowcursor_to_short_key(const RowCursor& key, size_t num_keys) {
```
Comment (zhannngchen) [inline be/src/olap/rowset/segment_v2/segment_iterator.h:171]: num_keys -> num_key_cols

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (wangbo) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:571]: 👌🏻

File: be/src/olap/rowset/segment_v2/segment_iterator.h (hunk 2 above)
Comment (wangbo) [reply be/src/olap/rowset/segment_v2/segment_iterator.h:287]: In the existing Doris semantics,  short key means prefix index, you can refer ```ShortKeyIndexDecoder```.
Then seek block means it is used to seek data page in Segment,
Binary search can be realized by comparing seek block and short key.

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (wangbo) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:571]: It stands for every time just seek one row.
If it is useless , clang compilation will failed.

File: be/src/olap/rowset/segment_v2/segment_iterator.h (hunk 3 above)
Comment (wangbo) [reply be/src/olap/rowset/segment_v2/segment_iterator.h:171]: What's the difference?

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**